app/views/doorkeeper/applications/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= form_for :application, url: oauth_applications_path(application), html: {class: 'form-horizontal', role: 'form'} do |f| %>
+<%= form_for :application, url: application.new_record? ? oauth_applications_path(application) : oauth_application_path(application), method: application.new_record? ? :post : :put, html: {class: 'form-horizontal', role: 'form'} do |f| %>
```

Comment:
For the URL yes, but our routes don't like the normal POST that happens, so seems like the method thing needs to stay.